addons/sale/models/sale.py

```diff
@@ -1313,7 +1313,7 @@ def _compute_untaxed_amount_invoiced(self):
                 `inv_line` is a customer invoice line linked to the SO line
                 `ref_line` is a customer credit note (refund) line linked to the SO line
         """
-        for line in self:
+        for line in self.filtered(lambda l: not l.display_type):
```

Comment:
When we did the task we tried to add business conditions instead of conditions on `display_type` if possible, so here if the problem is the missing currency, I would filter on that instead. So if other lines that are not section or note have the same issue, it will not crash either.

Reply:
Well, while I agree on the fact the filter should be based on a business condition, I don't agree it should not crash for lines of other kind than `display_type` sections/notes.

This is not normal to have a regular line without currency, and having a crash there will act as a safe-guard to prevent wrong computation.

Besides, regarding the performance strictly speaking, it's a bit counter-efficient to include in the loop and in the computation lines we know won't have any effect on the total or taxes.

Reply:
For the performance we could also argue to immediately exclude those where the price_subtotal is 0 then, which would also fix the section/notes.

But honestly I'm fine either way, this discussion only happens because the database is inconsistent which shouldn't be the case in the first place.

Reply:
> But honestly I'm fine either way, this discussion only happens because the database is inconsistent which shouldn't be the case in the first place.
On that I fully agree.

After a deeper thinking, adding the `filtered`, whether the condition is on `display_type`, `currency_id` or `price_subtotal` will be bad for performance, as it will add a O(n) complexity.

Besides, the current code is quite simple and its better keep it that way. I did not realize `currency_id` and `company_id` were related to `order_id`, and therefore the database is really inconsistent. We will just correct the data and let the code that way.

Thanks for the review :)